src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -833,8 +845,15 @@ private PostalAddress6 constructPostalAddress(
         String isoCountryCode;
         if (StringUtils.isNotBlank(countryCode)) {
             isoCountryCode = convertFIPSCountryCodeToISOCountryCode(countryCode);
+        } else if (stateAndPostalCodeAreWithinUnitedStates(templateCustomerProfile.getStateCode(),
+                templateCustomerProfile.getZipCode())) {
+            LOG.warn("constructPostalAddress, Customer Profile {} has a blank country code but seems to reference "
+                    + "a domestic address; defaulting to '{}'",
+                    templateCustomerProfile.getId(), KFSConstants.COUNTRY_CODE_UNITED_STATES);
+            isoCountryCode = KFSConstants.COUNTRY_CODE_UNITED_STATES;
```

Comment:
I think extracting this code to a method would add complexity to the log statement so I am okay with leaving it as is.